src/redux/books/books.js

```diff
@@ -1,18 +1,78 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 
-const bookSlice = createSlice({
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  const response = await axios.get('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/u77mSBCSxbBkVWREEpxM/books');
+  return response.data;
+});
+
+export const addBook = createAsyncThunk('books/addBook', async (book) => {
+  const response = await axios.post('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/u77mSBCSxbBkVWREEpxM/books', book);
+  return response.data;
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async (id) => {
+  const response = await axios.delete(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/u77mSBCSxbBkVWREEpxM/books/${id}`);
+  return response.data;
+});
+
+const booksSlice = createSlice({
   name: 'books',
-  initialState: [],
+  initialState: {
+    allBooks: [],
+    status: '',
+  },
   reducers: {
-    addBook: (state, action) => {
-      state.push(action.payload);
-    },
-    removeBook: (state, action) => {
-      state.filter((book) => book.id !== action.payload.id);
+    checkStatus: (state) => {
+      const newState = state;
+      newState.status = 'under construction';
     },
   },
```

Comment:
- This is amazing just the way you have used state. The code is also clean and precise